### PR TITLE
feat: sending a custom rdme user agent with all requests

### PR DESCRIPTION
**Pull request by erunion**
## 🧰 Changes

Much to my surprise and chagrin we weren't already sending a custom `rdme/(major).(minor).(patch)` user agent with all API requests. This fixes that!

## 🧬 QA & Testing

I've slightly shuffled around the `cleanHeaders` and `handleRes` libraries we had in `src/lib` into this new `fetch` wrapper in `src/lib/fetch`. Anything that once loaded `node-fetch` now loads this (which in turn loads `node-fetch`).

I've also added some unit tests for this to make sure that user agents are getting sent, changed every command test that mocks calls to our API to mock that API call + the user agent, and also cleaned up a few command tests that didn't have any explicit assertions (they were just silently testing either the mocks would succeed or that the command wouldn't fail).

### CODE REVIEW

File: __tests__/.eslintrc [hunk 1]
```diff
@@ -1,20 +1,3 @@
 {
-  "extends": ["@readme/eslint-config/testing"],
-  "env": {
-    "jest": true
-  },
-  "rules": {
-    "jest/expect-expect": [
```
Comment (erunion) [inline __tests__/.eslintrc:7]: All tests have explicit `expect()`s now so we don't need add these exclusions for nock.

File: __tests__/cmds/docs.test.js [hunk 2]
```diff
@@ -178,16 +181,24 @@ describe('rdme docs', () => {
...
-      });
+      await expect(docs.run({ folder: './__tests__/__fixtures__/new-docs', key, version })).resolves.toStrictEqual([
+        {
+          slug: 'new-doc',
+          body: '\nBody\n',
+          category: '5ae122e10fdf4e39bb34db6f',
+          title: 'This is the document title',
+          lastUpdatedHash: 'a23046c1e9d8ab47f8875ae7c5e429cb95be1c48',
+        },
+      ]);
```
Comment (kanadgupta) [inline __tests__/cmds/docs.test.js:197]: ugh we really had these tests in so many places lol 😭

File: __tests__/cmds/docs.test.js [hunk 3]
```diff
@@ -285,21 +296,29 @@ describe('rdme docs', () => {
...
         .reply(200, { version });
 
-      return docs.run({ folder: './__tests__/__fixtures__/slug-docs', key, version }).then(() => {
-        getMock.done();
-        postMock.done();
-        versionMock.done();
-      });
+      await expect(docs.run({ folder: './__tests__/__fixtures__/slug-docs', key, version })).resolves.toStrictEqual([
+        {
+          slug: 'marc-actually-wrote-a-test',
```
Comment (kanadgupta) [inline __tests__/cmds/docs.test.js:311]: lmao @mjcuva

File: __tests__/lib/commands.test.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jest/no-conditional-expect */
```
Comment (kanadgupta) [inline __tests__/lib/commands.test.js:1]: Could we disable it for each line or is it affecting the whole file?

File: src/lib/fetch.js [hunk 5]
```diff
@@ -1,10 +1,42 @@
...
+      'User-Agent': userAgent,
+    };
+  } else {
+    options.headers['User-Agent'] = userAgent;
+  }
+
+  return fetch(url, options);
+};
+
+module.exports.handleRes = async function handleRes(res) {
```
Comment (kanadgupta) [inline src/lib/fetch.js:24]: Could you add a li'l description comment above this function for consistency's sake?

**Review by kanadgupta — APPROVED** [review]
A couple of small notes but otherwise LGTM! The refactor is real nice 👏🏽 

Also quite bummed we hadn't been adding the user agent until now 😞

File: __tests__/lib/commands.test.js (hunk 4 above)
Comment (erunion) [reply __tests__/lib/commands.test.js:1]: Every test in this one has a conditional expect unfortunately.

**Pull request merged**